modules/rank-eval/src/main/java/org/elasticsearch/index/rankeval/RankEvalResponse.java

```diff
@@ -47,77 +46,60 @@
     public String toString() {
-        return "RankEvalResponse, quality: " + qualityLevel + ", unknown docs: " + unknownDocs;
+        return "RankEvalResponse, quality: " + qualityLevel + ", partial results: " + partialResults;
     }
 
     @Override
     public void writeTo(StreamOutput out) throws IOException {
         super.writeTo(out);
         out.writeDouble(qualityLevel);
-        out.writeVInt(unknownDocs.size());
-        for (String queryId : unknownDocs.keySet()) {
+        out.writeVInt(partialResults.size());
+        for (String queryId : partialResults.keySet()) {
             out.writeString(queryId);
-            Collection<RatedDocumentKey> collection = unknownDocs.get(queryId);
-            out.writeVInt(collection.size());
-            for (RatedDocumentKey key : collection) {
-                key.writeTo(out);
-            }
+            partialResults.get(queryId).writeTo(out);
         }
     }
 
     @Override
     public void readFrom(StreamInput in) throws IOException {
         super.readFrom(in);
         this.qualityLevel = in.readDouble();
-        int unknownDocumentSets = in.readVInt();
-        this.unknownDocs = new HashMap<>(unknownDocumentSets);
-        for (int i = 0; i < unknownDocumentSets; i++) {
+        int partialResultSize = in.readVInt();
+        this.partialResults = new HashMap<>(partialResultSize);
+        for (int i = 0; i < partialResultSize; i++) {
             String queryId = in.readString();
-            int numberUnknownDocs = in.readVInt();
-            Collection<RatedDocumentKey> collection = new ArrayList<>(numberUnknownDocs);
-            for (int d = 0; d < numberUnknownDocs; d++) {
-                collection.add(new RatedDocumentKey(in));
-            }
-            this.unknownDocs.put(queryId, collection);
+            EvalQueryQuality partial = new EvalQueryQuality(in);
+            this.partialResults.put(queryId, partial);
         }
     }
 
     @Override
     public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
         builder.startObject("rank_eval");
         builder.field("quality_level", qualityLevel);
-        builder.startObject("unknown_docs");
-        for (String key : unknownDocs.keySet()) {
-            Collection<RatedDocumentKey> keys = unknownDocs.get(key);
-            builder.startArray(key);
-            for (RatedDocumentKey docKey : keys) {
-                builder.startObject();
-                builder.field(RatedDocument.INDEX_FIELD.getPreferredName(), docKey.getIndex());
-                builder.field(RatedDocument.TYPE_FIELD.getPreferredName(), docKey.getType());
-                builder.field(RatedDocument.DOC_ID_FIELD.getPreferredName(), docKey.getDocID());
-                builder.endObject();
-            }
-            builder.endArray();
+        builder.startObject("details");
+        for (String key : partialResults.keySet()) {
+            partialResults.get(key).toXContent(builder, params);
```

Comment:
Makes, sense, I will rename this one now, we should also do a later round with this kind of cleanups.